test/notebooks/notebook_test_utils.py

```diff
@@ -139,109 +110,55 @@ def set_log_level_for_libraries(level=logging.WARNING):
         logging.getLogger(m).setLevel(level)
 
 
-@contextmanager
-def access_to_temp_onprem_secret_store(tmp_path: Path) -> Tuple[Path, str]:
-    """
-    Creates a temporary configuration store.
-    Brings up and subsequently destroys the Exasol Docker-DB.
-    Returns the temporary configuration store path and password.
-    """
-
-    # Create temporary secret store
-    store_path = tmp_path / 'tmp_config.sqlite'
-    # Should the password be constant or randomly generated is debatable.
-    # Strictly speaking, random password makes a test non-deterministic.
-    # On the other hand, it improves the security. If for some reason the
-    # temporary configuration store is not deleted, there will be no access
-    # to it since the password will be forgotten once the session is over.
-    store_password = generate_password(12)
-    secrets = Secrets(store_path, master_password=store_password)
-
-    # Set the configuration required by the ITDE manager and those the
-    # manager will not set after starting the Exasol Docker-DB.
-    _init_onprem_secret_store(secrets)
-
-    # Start the Exasol Docker-DB and then destroy it after the test finishes.
-    bring_itde_up(secrets)
-    try:
-        yield store_path, store_password
-    finally:
-        take_itde_down(secrets)
-
-
 @pytest.fixture(scope='session')
-def access_to_temp_saas_secret_store(tmp_path_factory) -> Tuple[Path, str]:
+def backend_setup(backend,
+                  saas_host,
+                  saas_pat,
+                  saas_account_id,
+                  database_name,
+                  backend_aware_onprem_database,
+                  backend_aware_saas_database_id,
```

Comment:
I guess using the id might avoid some of connection issues we had

Reply:
I understand the comment as to why we do not use the backend_aware_saas_database_id but use the database_name instead. That is to increase the code coverage. Include the case when the database_id is unknown and needs to be looked up using the name.

Reply:
ah ok, we actually use the name, might have overlooked that. Fine by me. We need to get the lookup anyway stable at somepoint.